dpcpp/matrix/csr_kernels.dp.cpp

```diff
@@ -207,10 +631,64 @@ void transpose_and_transform(std::shared_ptr<const DpcppExecutor> exec,
                              UnaryOperator op) GKO_NOT_IMPLEMENTED;
 
 
+template <bool conjugate, typename ValueType, typename IndexType>
+void generic_transpose(std::shared_ptr<const DpcppExecutor> exec,
+                       const matrix::Csr<ValueType, IndexType> *orig,
+                       matrix::Csr<ValueType, IndexType> *trans)
+{
+    const auto num_rows = orig->get_size()[0];
+    const auto num_cols = orig->get_size()[1];
+    auto queue = exec->get_queue();
+    const auto row_ptrs = orig->get_const_row_ptrs();
+    const auto cols = orig->get_const_col_idxs();
+    const auto vals = orig->get_const_values();
+
+    Array<IndexType> counts{exec, num_cols + 1};
+    auto tmp_counts = counts.get_data();
+    auto out_row_ptrs = trans->get_row_ptrs();
+    auto out_cols = trans->get_col_idxs();
+    auto out_vals = trans->get_values();
+    components::fill_array(exec, tmp_counts, num_cols, IndexType{});
```

Comment:
I think it would be better if we would use `queue->memset()` inside `components::fill_array` for DPC++ and keep the call here (IMO, the current code is more descriptive and I doubt we would get performance benefits from `memset()`).

Reply:
no, SYCL has a typed memset template